crates/common/src/result.rs

```diff
@@ -30,8 +30,9 @@ pub enum WasmErrorInner {
     /// AND wasm execution MUST immediately halt.
     /// The `Vec<u8>` holds the encoded data as though the guest had returned.
     HostShortCircuit(Vec<u8>),
-    /// Wasmer failed to compile machine code from wasm byte code.
-    Compile(String),
+    /// Wasmer failed to build a Module from wasm byte code.
+    /// With the feature `wasmer_sys` enabled, building a Module includes compiling the wasm.
+    ModuleBuild(String),
```

Comment:
Rename error type to clarify that it is an error arising when creating a new wasmer module. Only in `wasmer_sys` mode does compilation occur here.